### PR TITLE
Core: Migrating Util module to Junit5 

### PR DESCRIPTION
**Pull request by DaVincii**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 #7160 ".

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/util/TestInMemoryLockManager.java [hunk 1]
```diff
@@ -28,29 +30,26 @@
...
 public class TestInMemoryLockManager {
 
   private LockManagers.InMemoryLockManager lockManager;
   private String lockEntityId;
   private String ownerId;
 
-  @Rule public Timeout timeout = new Timeout(5, TimeUnit.SECONDS);
-
-  @Before
+  @Timeout(value = 5, unit = TimeUnit.SECONDS)
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestInMemoryLockManager.java:44]: should this be moved to the class level to be globally applied?

File: core/src/test/java/org/apache/iceberg/util/TestReachableFileUtil.java [hunk 2]
```diff
@@ -87,7 +83,7 @@ public void testManifestListLocations() {
     table.newAppend().appendFile(FILE_B).commit();
 
     List<String> manifestListPaths = ReachableFileUtil.manifestListLocations(table);
-    Assert.assertEquals(manifestListPaths.size(), 2);
+    assertThat(manifestListPaths.size()).isEqualTo(2);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestReachableFileUtil.java:86]:
```suggestion
    assertThat(manifestListPaths).hasSize(2);
```

File: core/src/test/java/org/apache/iceberg/util/TestReachableFileUtil.java [hunk 3]
```diff
@@ -119,7 +115,7 @@ public void testMetadataFileLocationsWithMissingFiles() {
     table.io().deleteFile(location);
 
     Set<String> metadataFileLocations = ReachableFileUtil.metadataFileLocations(table, true);
-    Assert.assertEquals(metadataFileLocations.size(), 2);
+    assertThat(metadataFileLocations.size()).isEqualTo(2);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestReachableFileUtil.java:118]: same as above. please also adjust the other places in this file

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 4]
```diff
@@ -43,28 +43,29 @@ public void testSingleRecord() {
...
-    Assert.assertEquals(0, map.size());
+    assertThat(map).isEmpty();
 
     map.put(record1, "1-aaa");
-    Assert.assertEquals(1, map.size());
-    Assert.assertFalse(map.isEmpty());
-    Assert.assertTrue(map.containsKey(record1));
-    Assert.assertTrue(map.containsValue("1-aaa"));
-    Assert.assertEquals("1-aaa", map.get(record1));
+    assertThat(map.size()).isOne();
```
[the remark this answers is not in the document]
Comment (nastra) [reply core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:49]: in fact this line can be removed and the check can be combined with the next one

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 5]
```diff
@@ -43,28 +43,29 @@ public void testSingleRecord() {
...
+    assertThat(map)
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:50]: no need to check the same thing multiple times:
```
assertThat(map)
        .hasSize(1)
        .containsEntry(record1, "1-aaa")
```

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 6]
```diff
@@ -43,28 +43,29 @@ public void testSingleRecord() {
...
+        .isNotEmpty()
+        .containsKey(record1)
+        .containsValue("1-aaa")
+        .containsEntry(record1, "1-aaa");
 
     Set<StructLike> keySet = map.keySet();
-    Assert.assertEquals(1, keySet.size());
-    Assert.assertTrue(keySet.contains(record1));
+    assertThat(keySet.size()).isOne();
+    assertThat(keySet).contains(record1);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:58]: can be combined to `assertThat(keySet).hasSize(1).contains(record1)`.
Please also do the same for all the other places

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 7]
```diff
@@ -43,28 +43,29 @@ public void testSingleRecord() {
...
 
     Collection<String> values = map.values();
-    Assert.assertEquals(1, values.size());
-    Assert.assertEquals("1-aaa", values.toArray(new String[0])[0]);
+    assertThat(values.size()).isOne();
+    assertThat(values.toArray(new String[0])[0]).isEqualTo("1-aaa");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:62]: why not `assertThat(values).hasSize(1).first().isEqualTo("1-aaa")`

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 8]
```diff
@@ -79,38 +80,37 @@ public void testMultipleRecord() {
...
-    Assert.assertTrue(map.containsKey(record3));
-    Assert.assertTrue(map.containsValue("1-aaa"));
-    Assert.assertTrue(map.containsValue("2-bbb"));
-    Assert.assertTrue(map.containsValue("3-null"));
-    Assert.assertEquals("1-aaa", map.get(record1));
-    Assert.assertEquals("2-bbb", map.get(record2));
-    Assert.assertEquals("3-null", map.get(record3));
+    assertThat(map)
+        .hasSize(3)
+        .containsKey(record1)
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:88]: we don't need to check for `containsKey` and `containsValue` if we already check for `containsEntry`

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 9]
```diff
@@ -120,29 +120,28 @@ public void testRemove() {
 
     Map<StructLike, String> map = StructLikeMap.create(STRUCT_TYPE);
     map.put(record, "1-aaa");
-    Assert.assertEquals(1, map.size());
-    Assert.assertEquals("1-aaa", map.get(record));
-    Assert.assertEquals("1-aaa", map.remove(record));
-    Assert.assertEquals(0, map.size());
+    assertThat(map.size()).isOne();
+    assertThat(map).containsEntry(record, "1-aaa");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:124]: `assertThat(map).hasSize(1).containsEntry(...)` and then you can remove the check above

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 10]
```diff
@@ -120,29 +120,28 @@ public void testRemove() {
...
   @Test
   public void testNullKeys() {
     Map<StructLike, String> map = StructLikeMap.create(STRUCT_TYPE);
-    Assert.assertFalse(map.containsKey(null));
+    assertThat(map).doesNotContainKey(null);
 
     map.put(null, "aaa");
-    Assert.assertTrue(map.containsKey(null));
-    Assert.assertEquals("aaa", map.get(null));
+    assertThat(map).containsKey(null).containsEntry(null, "aaa");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:138]: `containsKey` can be removed here

File: core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java [hunk 11]
```diff
@@ -155,13 +154,11 @@ public void testKeysWithNulls() {
...
     map.put(record2, "bbb");
 
-    Assert.assertEquals("aaa", map.get(record1));
-    Assert.assertEquals("bbb", map.get(record2));
+    assertThat(map).containsEntry(record1, "aaa").containsEntry(record2, "bbb");
 
     Record record3 = record1.copy();
-    Assert.assertTrue(map.containsKey(record3));
-    Assert.assertEquals("aaa", map.get(record3));
+    assertThat(map).containsKey(record3).containsEntry(record3, "aaa");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestStructLikeMap.java:160]: `containsKey` can be removed here

File: core/src/test/java/org/apache/iceberg/util/TestTableScanUtil.java [hunk 12]
```diff
@@ -125,8 +123,7 @@ public void testTaskGroupPlanning() {
 
     CloseableIterable<ScanTaskGroup<ParentTask>> taskGroups =
         TableScanUtil.planTaskGroups(CloseableIterable.withNoopClose(tasks), 128, 10, 4);
-
-    Assert.assertEquals("Must have 3 task groups", 3, Iterables.size(taskGroups));
+    Assertions.assertThat(taskGroups).as("Must have 3 task groups").hasSize(3);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestTableScanUtil.java:126]: why `Assertions.assertThat` here when you're using `assertThat` in all the other places in this file?

**Comment by skytin1004** [comment]
Hi @DaVincii, @nastra , It seems like this PR hasn't gotten any work done in days. there's also a branch conflict issue. If you're too busy to work on this, would you mind if I took over? I will fix the code based on the review and PR it again as a co-author.

**Comment by DaVincii** [comment]
> Hi @DaVincii, @nastra , It seems like this PR hasn't gotten any work done in days. there's also a branch conflict issue. If you're too busy to work on this, would you mind if I took over? I will fix the code based on the review and PR it again as a co-author.

@skytin1004 Got busy with some office work, now I will have some more time, will be updating it in next few days.

**Comment by DaVincii** [comment]
@nastra Updated the pull request with the review changes, please review.

File: core/src/test/java/org/apache/iceberg/util/TestJsonUtil.java [hunk 13]
```diff
@@ -18,247 +18,182 @@
...
 
-    Assertions.assertThatThrownBy(
+    assertThatThrownBy(
             () -> JsonUtil.getStringOrNull("x", JsonUtil.mapper().readTree("{\"x\": 23}")))
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Cannot parse to a string value: x: 23");
   }
 
-  @Test
-  public void getByteBufferOrNull() throws JsonProcessingException {
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestJsonUtil.java:173]: why is this test method and a few others being removed?

**Review by nastra — CHANGES_REQUESTED** [review]
there are tests being removed in `TestJsonUtil` (most likely from recent merge conflicts) that need to be restored

**Comment by DaVincii** [comment]
> there are tests being removed in `TestJsonUtil` (most likely from recent merge conflicts) that need to be restored

ah my bad, added back the test.

File: core/src/test/java/org/apache/iceberg/util/TestJsonUtil.java [hunk 14]
```diff
@@ -18,247 +18,202 @@
...
         .hasMessage("Cannot parse to a boolean value: x: \"true\"");
 
-    Assertions.assertThat(JsonUtil.getBool("x", JsonUtil.mapper().readTree("{\"x\": true}")))
-        .isTrue();
-    Assertions.assertThat(JsonUtil.getBool("x", JsonUtil.mapper().readTree("{\"x\": false}")))
-        .isFalse();
-  }
-
-  @Test
-  public void getIntArrayOrNull() throws JsonProcessingException {
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/util/TestJsonUtil.java:220]: this also needs to be reverted. In fact, can you please revert all changes to this file. I believe the only change that this file requires is switching from `org.junit.Test` to `org.junit.jupiter.api.Test` (everything else just adds too much visual noise

**Review by nastra — APPROVED** [review]
LGTM, thanks @DaVincii

**Comment by DaVincii** [comment]
> LGTM, thanks @DaVincii

I don't have write access, all the check have passed. Should be good to merge now.

**Pull request merged**